scripts/sdk-binary-size.gradle

```diff
@@ -0,0 +1,160 @@
+import groovy.json.JsonBuilder
+import groovy.json.JsonSlurper
+
+// Array of module names for evaluating SDK binary size
+final List<String> targetModules = ['tracking', 'messagingpush', 'messaginginapp']
+// Default output file name for the combined report
+final String defaultSdkReportName = 'sdk-binary-size.json'
+// Unit used for generating binary size report
+final String sdkSizeUnit = 'MB'
+
+// Task to generate combined report of binary sizes of all SDK modules
+// The report is generated in JSON format using apkscale plugin and is saved to
+// `$buildDir/sdk-binary-size.json` by default
+// This task can be run locally by executing `./gradlew generateSdkSizeReport`
+// Run `./gradlew generateSdkSizeReport -PwriteReportTo=<writeReportTo>` to save the report to
+// provided file path
+tasks.register("generateSdkSizeReport") {
+    // List all task dependencies for each library module
+    List<String> taskDependencies = new ArrayList<>()
+    // Add the assembleRelease task first for each library module as a dependency
+    // So that the apkscale can use the generated aar files to calculate the binary size
+    project.subprojects.findAll { Project subproject ->
+        subproject.plugins.hasPlugin('com.android.library')
```

Comment:
do we need to run for all library modules? or could we just run for `targetModules` ?

Reply:
We can run it for `targetModules` only. However, running it for all library modules gives me more confidence that APKScale will be able to locate all the AARs it requires. I lean towards running it for all library modules for this reason, but I'm flexible and have no strong preference as long as it works. If we believe running it only for `targetModules` could save time without causing issues, I'm happy to make that change and test it.

Reply:
I think we can make a more informed decision once we see the time taken by both approaches. Happy to go ahead with this and if we see its taking more time than we anticipate we can, revisit it.